colcon_ros/task/__init__.py

```diff
@@ -4,33 +4,14 @@
 import os
 
 
-def extend_cpp_with_app(args):
-    """Extend CMAKE_PREFIX_PATH with AMENT_PREFIX_PATH."""
+def append_app_to_cpp(env):
+    """Append AMENT_PREFIX_PATH to CMAKE_PREFIX_PATH."""
     ament_prefix_path = os.environ.get('AMENT_PREFIX_PATH')
```

Comment:
AFAIK, the hooks don't modify the `AMENT_PREFIX_PATH`, but should we get that from `env` just to be safe?

Reply:
I think getting it from `os.environ` is the right thing to do since you only want to append paths (underlay workspaces) which have been sourced before `colcon` was invoked. Paths from other packages in the same workspace are treated differently.

So even if packages would contribute to `get_command_environment` they shouldn't be added by this function.